### ISSUE TITLE
"num" argument

### ISSUE THREAD

**Issue by viehwegerlib** [dib-lab/sourmash#293]
If I use the API to create a signature, the `num` argument reflects the sketch size.

```
mh = MinHash(ksize=21, n=1234)
mh.add_sequence(some_sequence)
sig = signature.SourmashSignature(email='', minhash=mh)
sig._save()[3]['num']  # returns 1234
```

However, when I create a signature with `sourmash compute` it has e.g. `"num": 4294967295` in the `result.sig.json`. 

If I try to run e.g. `sourmash categorize`, it complains about this:

```
found 1 files to query
loaded query: ... (k=21, DNA)
...
  File "sourmash_lib/_minhash.pyx", line 320, in sourmash_lib._minhash.MinHash.similarity (sourmash_lib/_minhash.cpp:7382)
  File "sourmash_lib/_minhash.pyx", line 301, in sourmash_lib._minhash.MinHash.jaccard (sourmash_lib/_minhash.cpp:7196)
  File "sourmash_lib/_minhash.pyx", line 296, in sourmash_lib._minhash.MinHash.compare (sourmash_lib/_minhash.cpp:7001)
  File "sourmash_lib/_minhash.pyx", line 268, in sourmash_lib._minhash.MinHash.intersection (sourmash_lib/_minhash.cpp:6632)
TypeError: must have same num: 1234 != 4294967295
```

`sourmash gather` does work just fine.

So where does this number `4294967295` come from, i.e. how can I manually compute it to insert it into a "manually" constructed signature?

**Comment by viehwegerlib** [comment]
Is it related to [this code](https://github.com/dib-lab/sourmash/blob/ea1310da4695ba29ff6a74e9855c0c35a9c6b6c1/sourmash_lib/kmer_min_hash.hh):

```
KmerMinHash(unsigned int n, unsigned int k, bool prot, uint32_t s,
--
  | HashIntoType mx)
  | : // overflow num to represent "no maximum"
  | num(n > 0 ? n : -1),
  | ksize(k), is_protein(prot), seed(s),
  | // overflow max_hash to represent "no maximum", this simplifies
  | // the comparison in add_hash()
  | max_hash(mx > 0 ? mx : -1) {
  | if (n > 0) {
  | mins.reserve(num + 1);
  | }
  | // only reserve a finite amount of space for unbounded MinHashes
  | else {
  | mins.reserve(1000);
  | }
  | };
```
Also, some of the [test signatures](https://github.com/dib-lab/sourmash/blob/a9e49f6b68823f17179142512784a164d5cd67cf/tests/test-data/gather/GCF_000008545.1_ASM854v1_genomic.fna.gz.sig) have exactly the same `4294967295` number, so I guess this is some default value.

**Comment by ctb** [comment]
See #214. We plan to fix this :).

**Comment by ctb** [comment]
Fixed in #328! Now num=0 if you want to use scaled to collect all the hashes, otherwise it's the number of hashes you want to collect.
